frontend/www/js/omegaup/components/problem/creator/cases/AddPanel.test.ts

```diff
@@ -74,11 +78,111 @@ describe('AddPanel.vue', () => {
 
     expect(groups.length).toBe(1);
-    expect(groups[0].name).toBe('groupname');
+    expect(groups[0].name).toBe(testGroup);
     expect(groups[0].autoPoints).toBe(false);
     expect(groups[0].points).toBe(10);
     expect(groups[0].ungroupedCase).toBe(false);
+
+    const groupId = groups?.[0].groupID;
+
+    wrapper.setData({ tab: 'case' });
+    await Vue.nextTick();
+
+    const caseNameInput = wrapper.find('input[name="case-name"]');
+    const casePointsInput = wrapper.find('input[name="case-points"]');
+    const caseGroupName = wrapper.find('select[name="case-group"]');
+
+    await caseNameInput.setValue('groupedtestcase');
+    await casePointsInput.setValue(10);
+    await caseGroupName.setValue(groupId);
+
+    await wrapper.find('form').trigger('submit.prevent');
+
+    const groupedCases = store.state.casesStore.groups?.[0].cases;
+    expect(groupedCases.length).toBe(1);
+    expect(groupedCases[0].name).toBe('groupedtestcase');
+    expect(groupedCases[0].groupID).toBe(groupId);
+    expect(groupedCases[0].points).toBe(10);
   });
+
+  it('Should add an ungrouped case to the store', async () => {
+    const wrapper: Wrapper<AddPanel> = mount(AddPanel, {
+      localVue,
+      store: vuexStore,
+    });
+
+    wrapper.setData({ tab: 'case' });
+    await Vue.nextTick();
+
+    const nameInput = wrapper.find('input[name="case-name"]');
+    const pointsInput = wrapper.find('input[name="case-points"]');
+
+    await nameInput.setValue('testcase');
+    await pointsInput.setValue(50);
+
+    const updatedNameInput = wrapper.find('input[name="case-name"]')
+      .element as HTMLInputElement;
+    const updatedPointsInput = wrapper.find('input[name="case-points"]')
+      .element as HTMLInputElement;
+
+    expect(updatedNameInput.value).toBe('testcase');
+    expect(updatedPointsInput.value).toBe('50');
+
+    await wrapper.find('form').trigger('submit.prevent');
+
+    const store = wrapper.vm.$store as Store<StoreState>;
+    const groups = store.state.casesStore.groups;
+
+    expect(groups.length).toBe(1);
+    expect(groups[0].name).toBe('testcase');
+    expect(groups[0].autoPoints).toBe(false);
+    expect(groups[0].points).toBe(50);
+    expect(groups[0].ungroupedCase).toBe(true);
+  });
+
+  it('Should add multiple ungrouped cases to the store', async () => {
+    const wrapper: Wrapper<AddPanel> = mount(AddPanel, {
+      localVue,
+      store: vuexStore,
+    });
+
+    wrapper.setData({ tab: 'multiplecases' });
+    await Vue.nextTick();
+
+    const prefixInput = wrapper.find('input[name="multiple-cases-prefix"]');
+    const suffixInput = wrapper.find('input[name="multiple-cases-suffix"]');
+    const countInput = wrapper.find('input[name="multiple-cases-count"]');
+
+    await prefixInput.setValue('test');
+    await suffixInput.setValue('case');
+    await countInput.setValue(10);
+
+    const updatedpPrefixInput = wrapper.find(
+      'input[name="multiple-cases-prefix"]',
+    ).element as HTMLInputElement;
```

Comment:
In this case, you can keep the constants, but you can simplify the code a bit since you already found the element earlier. You can use it like this:
```suggestion
    const updatedpPrefixInput = prefixInput.element as HTMLInputElement;
```

Reply:
Hi @pabo99, following [this](https://github.com/omegaup/omegaup/pull/7651#discussion_r1640030470) comment, I have got rid of the `prefixInput` var. So, I just extracted them and checked again!